packages/@azure/template/package.json

```diff
@@ -17,7 +17,8 @@
     "build-browser": "tsc -p . && cross-env ONLY_BROWSER=true rollup -c",
     "build-node": "tsc -p . && cross-env ONLY_NODE=true rollup -c",
     "test": "mocha test-dist/**/*.js --reporter spec",
-    "pretest": "npm run build-test"
+    "pretest": "npm run build-test",
+    "eslint": "eslint src/**/*.ts test/**/*.ts -c .eslintrc.json"
```

Comment:
Do we want `eslint` to run during the CI builds, and cause the build to fail if there are problems?  If so, would `package.json` need additional changes to enable this behavior?

Reply:
Good point. If the build isn't failing, nobody would care about fixing the problems :)

I see most libraries (e.g., servicebus) use `tslint` in build script, while some use it in test script. The storage libraries don't have a script for it at all, even they all have a `tslint.json` config file.  I would prefer adding `npm run eslint` into the build task.

Another issue I've seen is that `eslint` seems to be much slower than `tslint` (~3x time).  Some libraries used to [exclude tests and samples from `tslint`](https://github.com/Azure/azure-event-hubs-node/blob/4a3961ee7165315b90dac974e94f305a1073b86b/client/package.json#L40).  I think they should have the same quality as the lib code thus should be linted as well.  But this will adds to the build time.

Reply:
Agree with adding `eslint` to build task.  For the template, how long does the build take with and without `eslint`?  I doubt the increased build time would have a meaningful impact on CI builds, though it might for inner-loop builds.  If so, we might want to enable `eslint` by default in CI builds but disable by default in inner-loop builds.

Reply:
for the template, `eslint` adds about 5 seconds. For event hubs I see `eslint` takes 15 seconds while `tslint` only takes ~5 seconds.

I don't like different settings between CI and inner-loop because it's frustrated to see it passing locally, but CI builds later failed. Maybe running it with `test` is not a bad idea, since the increased time isn't that significant when comparing to the test running time.

Reply:
CI uses three "phases" for each pipeline:

1. Build
2. Static Analysis
3. Test (in parallel across mulitple language versions and OS)

CI would ideally run `eslint` once during the "Static Analysis" phase, rather than repeating it N times during the "Test" phase.

For dev inner-loop, I agree running `eslint` as part of `test` makes sense.

I assume we can structure `package.json` to enable both the CI and inner-loop scenarios above?  If necessary, we can have different commands for CI and inner-loop testing.

Reply:
I hadn't thought about the scenario of N times during test. How about for now adding it to the `build` step, and if the increased build time becomes an annoying issue we can tweak more?

Reply:
For CI we are pretty strict about keeping the "Build" phase as fast as possible, so we would only want to run `eslint` during the "Static Analysis" phase.  For inner-loop I defer to the people working in the repo every day -- whatever you think is the best fit for your workflow(s).

So for CI, we just need some way to run `build` without `eslint`.

Reply:
Do we have a standardized/conventional npm step for "Static Analysis" where we could add `eslint`? I could add `eslint` to `test`, and another `test-no-lint` step for CI.

Reply:
Currently, CI invokes command `npm run audit-client`, which points here in the root `package.json`:

https://github.com/Azure/azure-sdk-for-js/blob/master/package.json#L60-L62

We could either bundle `eslint` with `npm audit`, or create a new command in the root `package.json` named something like `analyze-client`, which would run both `npm audit` and `eslint`.

Reply:
> something like `analyze-client`, which would run both `npm audit` and `eslint`.

I like the idea and updated the PR.